gate.py

```
@@ -0,0 +1,348 @@
+    if not seed:
+        return RedirectResponse("/gate", status_code=303)
+
+    question = _get_question_for_position(n, seed)
+    if not question:
+        return RedirectResponse("/gate", status_code=303)
+
+    answer_text = answer.strip()
+
+    # Atomic read-modify-write with row lock to prevent stale overwrites
+    async with pool.acquire() as conn:
+        async with conn.transaction():
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)